packages/shared/src/baduk/badukWithAbstractBoard.ts

```diff
@@ -0,0 +1,269 @@
+import { AbstractGame } from "../abstract_game";
+import type { MovesType } from "../abstract_game";
+import type { IBadukBoard } from "../Interfaces/IBadukBoard";
+import { BadukBoardAbstract } from "./board/BadukBoardAbstract";
+import type { Intersection } from "./board/intersection";
+
+export enum Color {
+    EMPTY = 0,
+    BLACK = 1,
+    WHITE = 2,
+}
+
+export enum BoardPattern {
+    Uknown = 0,
+    Rectangular = 1,
+    Polygonal = 2,
+}
+
+export interface BadukWithAbstractBoardConfig {
+    width: number;
+    height: number;
+    komi: number;
+    pattern: BoardPattern;
+}
+
+export interface BadukWithAbstractBoardState {
+    board: IBadukBoard;
```

Comment:
A note on `AbstractGame.exportState`:

The goal for that is to be able to store game state in a relatively minimal way (and should be JSON friendly since we may pass it via API save it in the database).  I don't think we necessarily need to store information about edges or the physical position since that can all be derived from the config.

export interface BadukWithAbstractBoardState {
    intersections: Color[];
        ...
}

exportState() {
    return {
        board: this.Intersections.map((intersection) => intersection.StoneState.Color),
            ...
    }
}